src/commands/snipe.command.ts

```diff
@@ -0,0 +1,222 @@
+/*
+  Remoji - Discord emoji manager bot
+  Copyright (C) 2021 Shino <shinotheshino@gmail.com>.
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Affero General Public License as published
+  by the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { SlashCommand, SlashCreator, CommandContext, CommandOptionType, Permissions } from "slash-create";
+
+import { getRemainingGuildEmoteSlots, EmbedUtil, getEmoteCDNLink, arraySumColumn } from "../lib/utils";
+import { Bot } from "../lib/bot";
+import logger from "../lib/logger";
+import { stripIndents } from "common-tags";
+import Constants from "../Constants";
+import got from "got";
+
+export default class SnipeCommand extends SlashCommand {
+  constructor(creator: SlashCreator) {
+    super(creator, {
+      name: "snipe",
+      description: "Snipe emojis from a message link",
+      options: [
+        {
+          name: "url",
+          description: "The message URL to snipe emojis from",
+          type: CommandOptionType.STRING,
+          required: true,
+        },
+      ],
+      guildIDs: ["683009037830324235"],
+    });
+    this.filePath = __filename;
+  }
+
+  async run(ctx: CommandContext): Promise<void | string> {
+    if (!ctx.guildID) {
+      await ctx.send({ embeds: [EmbedUtil.error(":x: This command may only be used in a server.")] });
+      return;
+    }
+
+    if (!ctx.member?.permissions.has(Permissions.FLAGS.MANAGE_EMOJIS)) {
+      await ctx.send({ ephemeral: true, embeds: [EmbedUtil.error(":lock: You need the Manage Emojis permission to use this command.")] });
+      return;
+    }
+
+    const url = ctx.options.url as string;
+
+    const regex = /^https:\/\/(canary\.|ptb\.)?discord\.com\/channels\/\d{17,20}\/(?<channel>\d{17,20})\/(?<message>\d{17,20})$/;
+    const groups = url.match(regex)?.groups;
+
+    if (!groups) {
+      await ctx.send({
+        ephemeral: true,
+        embeds: [EmbedUtil.error(":x: That doesn't look like a valid message URL.")],
+      });
+      return;
+    }
+
+    const bot = Bot.getInstance().client;
+
+    const message = await bot.getMessage(groups.channel, groups.message);
```

Comment:
```suggestion
    const message = await client.getMessage(groups.channel, groups.message);
```

Reply:
Should be wrapped in `try`/`catch` in case bot doesn't have access -- or, check for guild, then channel, then message in order.